Framework/PythonInterface/plugins/algorithms/CreatePoleFigureTableWorkspace.py

```diff
@@ -0,0 +1,231 @@
+        peak_param_ws = self.getProperty("PeakParameterWorkspace").value
+        hkl = self.getProperty("Reflection").value
+        hkl_is_set = not self.getProperty("Reflection").isDefault
+        chi_thresh = self.getProperty("Chi2Threshold").value
+        x0_thresh = self.getProperty("PeakPositionThreshold").value
+
+        # generate a detector table to get detector positions
+        det_table = self.exec_child_alg(
+            "CreateDetectorTable", InputWorkspace=ws, IncludeDetectorPosition=True, DetectorTableWorkspace="det_table"
+        )
+        det_pos = np.asarray(det_table.column("Position"))
+
+        # get the source pos for ki
+        source_pos = np.asarray(ws.getInstrument().getSource().getPos())
+
+        # get the rotation matrix of the sample
+        rot_mat = ws.run().getGoniometer().getR()
+
+        # matrix to put detectors in sample frame
+        to_pole_view = np.linalg.inv(rot_mat)
+
+        # get Qs
+        qds = det_pos / np.linalg.norm(det_pos, axis=1)[:, None]
+        # normalise source pos then take away from sample pos for ki
+        qi = np.zeros(3) - source_pos / np.linalg.norm(source_pos)
+        Qs = qds - qi
+        Qs = Qs / np.linalg.norm(Qs, axis=1)[:, None]
+
+        # put detectors in sample reference frame
+        sample_frame_Qs = to_pole_view @ Qs.T
+
+        # calculate alpha and beta for the Qs in the sample ref frame
+        ab_arr = _get_alpha_beta_from_cart(sample_frame_Qs)
+
+        if peak_param_ws:
+            # assume that the peak_param_ws has only the peak of interest fitted
+            intensities = np.asarray(peak_param_ws.column("I"))
+            if not self.no_x0_needed:
+                x0s = np.asarray(peak_param_ws.column("X0"))
+            else:
+                x0s = np.ones((len(ab_arr)))
+            if not self.no_chi2_needed:
+                chi2 = np.asarray(peak_param_ws.column("chi2"))
+            else:
+                chi2 = np.zeros((len(ab_arr)))
+
+        else:
+            intensities = np.ones((len(ab_arr)))
+            chi2 = np.zeros((len(ab_arr)))
+            x0s = np.ones((len(ab_arr)))
+
+        if hkl_is_set:
+            hkl = _parse_hkl(hkl)
+            xtal = ws.sample().getCrystalStructure()
+            peak = xtal.getUnitCell().d(hkl)
+            scat_power = _calc_scattering_power(xtal, hkl)
+        else:
+            # if no hkl provided, peak will be set to mean x0 and scattering power will be 1
+            peak = np.mean(x0s)
+            scat_power = 1.0
+
+        # create the table to hold the data
+        table_ws = self.exec_child_alg("CreateEmptyTableWorkspace", OutputWorkspace="_tmp")
+        table_ws.addColumn(type="double", name="Alpha", plottype=2)
+        table_ws.addColumn(type="double", name="Beta", plottype=2)
+        table_ws.addColumn(type="double", name="Intensity", plottype=2)
+
+        # go through each detector group
+        for index, (a, b) in enumerate(ab_arr):
```

Comment:
This is really nit-picking sorry, I wouldn't mention it if there weren't any other changes! But I think `index` is an uninformative variable name - obviously it's the index of the rows in the `ab_arr` - but it might be possible to indicate here what a row in the `ab_arr` corresponds to? Would `igroup` or at least `irow` etc. be better? Where the prefex `i` normally indicates an index (at least to me). Feel free to disagree or suggest alternatives!